### PR TITLE
Mine page: Create views and templates

### PR DESCRIPTION
**Pull request by ShellyGolden**
image

### CODE REVIEW

File: house/views.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
+from house.constants import HOME_PAGE_ROUTE, HOUSE_PAGE_ROUTE
+from .models import House
+from expenses.models import Expenses
+from django.shortcuts import render, get_object_or_404
 
 
 def home_page(request):
-    raise NotImplementedError
+    # this line is needed till the login of the mine page will create and merge
+    print(House.objects.all().order_by("?").first().house_id)
```
Comment (ronyyosef) [inline house/views.py:9]: Please remove prints before merging it into master
You did it to take a random house id and then go to its URL?

File: house/views.py [hunk 2]
```diff
@@ -13,7 +19,12 @@ def house_login(request):
 
 
 def house_view(request, house_id):
-    raise NotImplementedError
+    house = get_object_or_404(House, pk=house_id)
+    expenses_list = Expenses.objects.filter(house_name=house)
```
Comment (ronyyosef) [inline house/views.py:23]: Hey, I think there is a better way to do that in Django `Expenses.objects.filter(house_name=house)` 
Please take a look here: https://docs.djangoproject.com/en/4.0/topics/db/queries/#following-relationships-backward
Or here:  https://stackoverflow.com/questions/34184046/how-to-get-all-objects-referenced-as-foreignkey-from-given-field-in-a-module-in

File: templates/house_view/mine_page.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{% extends "base.html" %} {%load static%} {%block content%}
+<link rel="stylesheet" type="text/css" href="{% static 'css/mine_house_and_expenses.css' %}" />
+<body>
+  <div class="expenses_table-container">{% include 'house_view/expenses_table.html' %}</div>
+  <div class="house_table-container">{% include 'house_view/house_table.html' %}</div>
+  <div class="add-expense-container">{% include 'house_view/add_expense.html' %}</div>
+</body>
+{% endblock %}
```
Comment (ronyyosef) [inline templates/house_view/mine_page.html:8]: Missing empty line at the end of the file

File: templates/house_view/expenses_table.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      <td>{{ expense.description }}</td>
+    </tr>
+    {% empty %}
+    <tr>
+      <td colspan="4">No expense in the database.</td>
+    </tr>
+    {% endfor %}
+  </tbody>
+</table>
+{%endblock table%}
```
Comment (ronyyosef) [inline templates/house_view/expenses_table.html:29]: Missing empty line at the end of the file

**Review by ronyyosef — CHANGES_REQUESTED** [review]
Well done!
I left some comments, update me if you have any questions.
Also, please add a bit of styling to the tables and the look of the page if you can, you can use bootstrap to do it easily
Are we going to work on adding more things to that page? We can add the rest in different PR, just make the structure of the page clear here so later on we will not need to change the entire structure of the page.
 At the start we talk about that page looking something like this:
image

File: house/views.py (hunk 1 above)
Comment (ShellyGolden) [reply house/views.py:9]: as I wrote - I'll remove the print line after @LiorKGOW will do the login view.

**Comment by ShellyGolden** [comment]
about the design - 
for now, we have only the house details and the expenses of the house. 
Maybe in the future we will add the statistics data. 
as I wrote - Ill think about a better way to show the data :) 
If you have any idea feel free to write to me
thanks@
@ronyyosef

**Review by LiorKGOW — COMMENTED** [review]
Overall nice work @ShellyGolden 
I had a question about the house/views.py, added a comment there

Also, Have a look on what Shay commented on the DFD, maybe we should add an option of editing an expense
I have started a discussion about it on Slack

File: house/urls.py [hunk 5]
```diff
@@ -7,6 +7,6 @@
     path('', views.home_page, name='home'),
     path('global/', views.global_page, name='global_page'),
     path('login', views.house_login, name='house_login'),
-    path('<int:house_id>/', views.house_view, name='house_view'),
+    path('<str:house_id>/', views.house_view, name='house_view'),
```
[the remark this answers is not in the document]
Comment (ShellyGolden) [reply house/urls.py:10]: this is the type of the house_id.
@LiorKGOW

**Comment by adiabramovitch** [comment]
@ShellyGolden Is something missing? did all the conversations were resolved? why are we on draft mode?

**Comment by ShellyGolden** [comment]
> 

I want to edit the UI. 
I'll do it today, and then I'll remove the draft mode
@adiabramovitch

File: templates/house_view/mine_page.html (hunk 3 above)
Comment (ShellyGolden) [reply templates/house_view/mine_page.html]: fixed

File: templates/house_view/expenses_table.html (hunk 4 above)
Comment (ShellyGolden) [reply templates/house_view/expenses_table.html]: fixed

File: house/views.py (hunk 2 above)
Comment (ShellyGolden) [reply house/views.py:23]: for now, I'll leave it as it is, 
I'll take a look again
thanks
@ronyyosef

**Comment by ShellyGolden** [comment]
Hi
I updated the screenshot with the newest version of code.
please take a look

**Comment by ronyyosef** [comment]
I think you can  squish the two commits into one
The first one doesn't really means anything

**Review by ronyyosef — APPROVED** [review]
Well done!
Where will the button to add and edit expenses will be show?
Same avout edit house

**Comment by ShellyGolden** [comment]
> Well done!
> 
> Where will the button to add and edit expenses will be show?
> 
> Same avout edit house

I think that they will be above the expenses table. 
Ill think about it 🙂

File: house/views.py [hunk 6]
```diff
@@ -1,21 +1,29 @@
-from django.shortcuts import render
-from house.constants import HOME_PAGE_ROUTE, GLOBAL_PAGE_ROUTE
+from house.constants import HOME_PAGE_ROUTE, HOUSE_PAGE_ROUTE
+from .models import House
```
Comment (LiorKGOW) [inline house/views.py:2]: why did you write here .models and not house.models?
and on the next line expenses.models?

Don't they need to have the same syntax?

File: house/views.py [hunk 7]
```diff
@@ -1,21 +1,29 @@
...
+    raise NotImplementedError
 
 
 def house_login(request):
     raise NotImplementedError
 
 
 def house_view(request, house_id):
-    raise NotImplementedError
+    house = get_object_or_404(House, pk=house_id)
```
Comment (LiorKGOW) [inline house/views.py:22]: I read about get_object_or_404
nice use 👍 
I think it could be appropriate in the login page as well, I'll try adding it

**Review by LiorKGOW — COMMENTED** [review]
nice work @ShellyGolden 
I left some comments
pls have a look

about the first commit, you could open a new PR with the updated main
and then add your changes of the second commit
after that you'll be left with only one commit with your changes

I had trouble merging it too, and that solved the extra meaningless commit problem for me

File: house/views.py (hunk 7 above)
Comment (ronyyosef) [reply house/views.py:22]: I don't think we should use it in the login, If a user has some errors with the login we need to display him the error and not send him to 404. for example display him "there is no house with **** house_id"
@LiorKGOW

**Comment by adiabramovitch** [comment]
What about tests?

**Comment by ShellyGolden** [comment]
> What about tests?

I don't have logic in this PR to test.
I'll add tests in the PR with the logic of add/edit expense

File: house/views.py (hunk 6 above)
Comment (ShellyGolden) [reply house/views.py:2]: this is the house/views, so I'm already in the house model. 
and the expenses model is a separate model

**Comment by ShellyGolden** [comment]
> > What about tests?
> 
> I don't have logic in this PR to test. I'll add tests in the PR with the logic of add/edit expense

Does I need to add?
@adiabramovitch

**Review by LiorKGOW — APPROVED** [review]

**Comment by ShellyGolden** [comment]
> What about tests?

I'll add today

**Review by ronyyosef — APPROVED** [review]

**Review by orIsrael — APPROVED** [review]

File: house/tests/house_tests.py [hunk 8]
```diff
@@ -36,6 +41,12 @@ def generate_house(generate_country, generate_city):
     return house
 
 
+@pytest.fixture
+def generate_get_request(request_factory):
```
Comment (Yoavsc129) [inline house/tests/house_tests.py:45]: This fixture is unused. You can look at my PR https://github.com/redhat-beyond/Xpense/pull/71 to see the relevant test (test_board_function)

**Review by Yoavsc129 — CHANGES_REQUESTED** [review]

File: house/constants.py [hunk 9]
```diff
@@ -1,2 +1,7 @@
 HOME_PAGE_ROUTE = 'home.html'
 GLOBAL_PAGE_ROUTE = 'global/global.html'
+MINE_EXPENSES_TITLE_ROUTE = 'house_view/expenses_table.html'
+MINE_HOUSE_TABLE_ROUTE = 'house_view/house_table.html'
+MINE_HOUSE_TABLE_TITLE_ROUTE = 'house_view/house_table_title.html'
+MINE_MINE_PAGE_ROUTE = 'house_view/mine_page.html'
```
Comment (LiorKGOW) [inline house/constants.py:6]: Could you change it to just 1 Mine ? 😇

File: house/tests/house_tests.py [hunk 10]
```diff
@@ -1,5 +1,10 @@
 import pytest
 from house.models import House, City, Country
+from django.template.loader import get_template
+from django.template import TemplateDoesNotExist
+from house.constants import MINE_EXPENSES_TITLE_ROUTE, MINE_HOUSE_TABLE_ROUTE, MINE_HOUSE_TABLE_TITLE_ROUTE
+from house.constants import MINE_MINE_PAGE_ROUTE, MINE_SIDEBAR_ROUTE
```
Comment (LiorKGOW) [inline house/tests/house_tests.py:6]: Same here the const URL, and you could also make only one line of this import

File: house/tests/house_tests.py [hunk 11]
```diff
@@ -58,3 +69,32 @@ def test_delete_house(self, generate_house):
...
+
+        try:
+            get_template(MINE_HOUSE_TABLE_TITLE_ROUTE)
+        except TemplateDoesNotExist:
+            assert False, f"Template {MINE_HOUSE_TABLE_TITLE_ROUTE} does not exist"
+
+        try:
+            get_template(MINE_MINE_PAGE_ROUTE)
+        except TemplateDoesNotExist:
+            assert False, f"Template {MINE_MINE_PAGE_ROUTE} does not exist"
```
Comment (LiorKGOW) [inline house/tests/house_tests.py:95]: Same here like the comment in the const URL

File: house/views.py [hunk 12]
```diff
@@ -1,7 +1,7 @@
+from house.constants import HOME_PAGE_ROUTE, MINE_MINE_PAGE_ROUTE, GLOBAL_PAGE_ROUTE
```
Comment (LiorKGOW) [inline house/views.py:1]: Same here the const URL

File: house/views.py [hunk 13]
```diff
@@ -24,7 +24,11 @@ def house_login(request):
 
 
 def house_view(request, house_id):
-    raise NotImplementedError
+    house = get_object_or_404(House, pk=house_id)
+    expenses_list = Expenses.objects.filter(house_name=house)
+    context = {'house': house,
+               'house_expenses': expenses_list}
+    return render(request, MINE_MINE_PAGE_ROUTE, context)
```
Comment (LiorKGOW) [inline house/views.py:31]: Same here

**Review by LiorKGOW — CHANGES_REQUESTED** [review]
Hey @ShellyGolden left you some comments on the updated PR
GJ 👍

File: house/constants.py (hunk 9 above)
Comment (ShellyGolden) [reply house/constants.py:6]: I wanted to classify all the MINE constants that would start with the same word @LiorKGOW

File: house/tests/house_tests.py (hunk 10 above)
Comment (ShellyGolden) [reply house/tests/house_tests.py:6]: Because of the length of the line I split the lines @LiorKGOW

File: house/tests/house_tests.py (hunk 11 above)
Comment (ShellyGolden) [reply house/tests/house_tests.py:95]: I answered you in the first response

File: house/views.py (hunk 12 above)
Comment (ShellyGolden) [reply house/views.py:1]: I answered you in the first response

**Review by LiorKGOW — APPROVED** [review]
👍

File: house/tests/house_tests.py (hunk 8 above)
Comment (ShellyGolden) [reply house/tests/house_tests.py]: thanks!
fixed

**Comment by ShellyGolden** [comment]
I think that the PR is ready to merge
@adiabramovitch @svatares

**Review by Yoavsc129 — APPROVED** [review]

**Review by svatares — APPROVED** [review]

**Pull request merged**